crates/red_knot_python_semantic/resources/mdtest/subscript/string.md

```diff
@@ -20,11 +23,10 @@ reveal_type(b)  # revealed: Unknown
 ## Function return
 
 ```py
-def add(x: int, y: int) -> int:
-    return x + y
+def int_instance() -> int: ...
 
 
-a = "abcde"[add(0, 1)]
+a = "abcde"[int_instance()]
 # TODO: Support overloads... Should be `str`
```

Comment:
I understand that we want to infer `str` here. But I don't understand the "Support overloads" comment. Can someone explain?

Reply:
We attempt to infer the type here by looking up [`str.__getitem__` in typeshed](https://github.com/python/typeshed/blob/c417573eb6f59c8f40b7c0dc7b1d75549b8f6e59/stdlib/builtins.pyi#L592) and looking at what that method returns. But `str.__getitem__` is an [overloaded function](https://docs.python.org/3/library/typing.html#typing.overload) in typeshed, and we don't yet have the ability to understand those.

In fact, we just infer `@Todo` as the return type for all decorated functions currently, since the function (and its return type) is transformed by the decorator before the user sees it, so it's less simple than just "believe the return type" when it comes to decorated functions.

Reply:
Thanks!